### ISSUE TITLE
Surface Pro 4: No battery status after patch

### ISSUE THREAD

**Issue by xinyuwang1209** [jakeday/linux-surface#273]
the battery information disappeared in all status bars (I tried i3blocks and conky) and command acpi does not return anything

uname -r returns

```
4.18.7-1-surface
```
dmesg | grep acpi returns
```
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x01] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x02] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x03] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x04] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x05] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x06] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x07] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x08] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x09] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x0a] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x0b] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x0c] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x0d] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x0e] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x0f] high edge lint[0x1])
[    0.000000] ACPI: LAPIC_NMI (acpi_id[0x10] high edge lint[0x1])
[    0.080054] acpiphp: ACPI Hot Plug PCI Controller Driver version: 0.5
[    0.180566] acpi PNP0A08:00: _OSC: OS supports [ExtendedConfig Segments MSI]
[    0.180678] acpi PNP0A08:00: PCIe port services disabled; not requesting _OSC control
[    0.517373] clocksource: acpi_pm: mask: 0xffffff max_cycles: 0xffffff, max_idle_ns: 2085701024 ns
[    1.834230] surface_acpi: Microsoft Surface ACPI Driver version 0.1
[    1.834260] surface_acpi: Attaching device
[  280.923282] surface_acpi: Event received 81
[  282.099253] surface_acpi: Event received 81
[  282.099258] surface_acpi: Event received 80
[  282.103198] surface_acpi: Event received 80
[  292.290616] surface_acpi: Event received 81
[  292.290620] surface_acpi: Event received 80
[  292.295676] surface_acpi: Event received 80
[  305.668353] surface_acpi: Event received 81
[  305.668357] surface_acpi: Event received 80
[  305.672239] surface_acpi: Event received 80
[  307.709935] surface_acpi: Event received 81
[  307.709938] surface_acpi: Event received 80
[  307.714917] surface_acpi: Event received 80
[  315.890770] surface_acpi: Event received 81
[  315.890773] surface_acpi: Event received 80
...
```

The message I got from dmesg is different from issue #28

**Comment by xinyuwang1209** [comment]
Also here is my acpidump file:
https://github.com/xinyuwang1209/acpi/blob/master/acpidata.dat
I am really appreciated if anyone can help.

**Comment by ryanerwin** [comment]
I'm having the same issue on the Surface Pro 4.

```
$ uname -r
4.18.16-arch1-1-surface

$ ls -l /sys/class/power_supply
total 0
drwxr-xr-x 3 root root 0 Nov  9 16:32 ADP1/

$ acpi -b
# no output

$ sudo tlp stat|grep -i batter
RADEON_DPM_STATE_ON_BAT=battery
+++ Battery Status
No batteries detected.

$ acpi -V
Adapter 0: off-line

$ upower -i /org/freedesktop/UPower/devices/battery_BAT0
  native-path:          (null)
  power supply:         no
  updated:              Thu 01 Jan 1970 08:00:00 AM +08 (1541758148 seconds ago)
  has history:          no
  has statistics:       no
  unknown
    warning-level:       unknown
    battery-level:       unknown
    icon-name:          '(null)'
```

I'm not sure if it's related to:
Surface Pro 2017 / Book 2 / Laptop: Battery Status Not Working #28 https://github.com/jakeday/linux-surface/issues/28
Or:
Surface Pro 4: Battery sometimes still set as AC power even when discharging #223 https://github.com/jakeday/linux-surface/issues/223

Both of which are open...

There's a script called  `mshw0084-rqst.py`: Surface Book 2 / Surface Pro (2017) / Surface Laptop UART protocol proof-of-concept script that seems to be working for new Surfaces and SB2s, but I can't get it to work with the SP4...
https://gist.github.com/qzed/01a93568efb863f1b7887f0f375c03fc

**Comment by qzed** [comment]
The approach I'm working on for the 5th gen. Surface devices won't work here, the ACPI/DSDT is just too different. If it works on the upstream kernel I suggest you try to compile a kernel without the ACPI-related patches, via this you could then figure out where the issue comes from. I don't own a SP4, so I can't really help you guys here, sorry.

**Comment by qzed** [comment]
Interesting, I think there might be a driver in the ACPI patch that attaches to a device and via that prevents the proper upstream driver to load. My initial guess would be that it's the one attaching to `PNP0C0A` which would be a generic battery device (I can't see any special driver for the SP4 battery upstream, so this assumption probably holds). If I'm correct this should get resolved as a side-effect of the battery patches I'm currently working on for the other devices.

**Comment by cs-ktonkovic** [comment]
Can confirm this bug on 4.18.8-1-surface4 and Manjaro.

**Comment by yuzvir** [comment]
> Can confirm this bug on 4.18.8-1-surface4 and Manjaro.

There is a linux-surface4 package in aur for arch / manjaro. It works pretty well for me and has working battery status.

**Comment by qzed** [comment]
Should be fixed as side-effect of #321. If someone already wants to test that the PR really fixes the issue, try [this](https://github.com/qzed/linux-surface/releases/tag/v4.18.20-pre2) kernel.
